tests/integration/synapseclient/core/test_version_check.py

```diff
@@ -0,0 +1,20 @@
+"""Integration tests for version checking"""
+
+from synapseclient.core.version_check import _get_version_info_from_pypi, version_check
+
+
+async def test_version_check():
+    """Integration checks for version_check"""
+    # Check current version against pypi version file
+    version_check()
+
+    # Should be higher than current version and return true
+    assert version_check(current_version="999.999.999")
+
+    # Test out of date version
+    assert not version_check(current_version="0.0.1")
```

Comment:
Could you add one additional item to the assertions to verify that an HTTP call was fired off to pypi to get the version?
It'll be similar to how these SPYs work: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/8663e459b321270b65a0305c5126b0e955648fc9/tests/integration/synapseclient/core/test_download.py#L98-L99

Reply:
Is this enough?

```
    # Assert _get_version_info_from_pypi called  when running version_check
    spy = mocker.spy(synapseclient.core.version_check, "_get_version_info_from_pypi")
    version_check()
    spy.assert_called_once()
```
The test you linked has quite a bit going on, but it mostly concerns checking the downloaded file.

Reply:
In this case you'd want to add a `mocker.spy` for the `httpx.get` function call. The verification would be to make sure that the HTTP URL you are expecting (The PYPI URL) was called.

Reply:
Like:

```
    # Assert httpx.get called  when running version_check
    spy = mocker.spy(httpx, "get")
    version_check()
    spy.assert_called_once()
```

Reply:
Yeah. The only thing is that you'd use `assert_called_once_with`

Reply:
done